x/dsmr/storage.go

```diff
@@ -40,10 +41,9 @@ func (NoVerifier[T]) Verify(Chunk[T]) error {
 }
 
 type StoredChunkSignature[T Tx] struct {
-	Chunk          Chunk[T]
-	LocalSignature NoVerifyChunkSignatureShare // Decouple signature share / certificate types
-	Cert           *ChunkCertificate
-	Available      bool
+	Chunk     Chunk[T]
+	Cert      *ChunkCertificate[T]
+	Available bool
```

Comment:
(unrelated to this PR) What's the intent of `Available` ? Why would we need this flag rather than just returning the chunk if it's available?

Reply:
The rationale was that we wouldn't serve `GetChunk` requests unless the chunk was considered "available". That being said... I'm not sure why I added this flag at all because we only store chunks after we've already generated a signature so any chunk written is guaranteed to be available.

Reply:
I think we can remove, but don't need to in this PR